ner_v2/detectors/textual/utils.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import absolute_import
+
+import json
+import six
+
+from chatbot_ner.config import ner_logger
+from language_utilities.constant import ENGLISH_LANG
+
+from ner_constants import (DATASTORE_VERIFIED, MODEL_VERIFIED,
+                           FROM_FALLBACK_VALUE, ORIGINAL_TEXT, ENTITY_VALUE, DETECTION_METHOD,
+                           DETECTION_LANGUAGE, ENTITY_VALUE_DICT_KEY)
+from ner_v2.detectors.textual.text_detection import TextDetector
+
+
+def verify_text_request(request):
+    """
+    Check the request object
+    1. If proper message or entity is present in required
+    format.
+
+    2. If length of message or entity is in allowed range
+
+    Args:
+        request: API request object
+
+    Returns:
+        Raises KeyError if message or entities are not present
+        Raises TypeError if message is not list or entities is not dict type
+        Else Return none
+    """
+
+    request_data = json.loads(request.body)
+    message = request_data.get("message")
+    entities = request_data.get("entities")
+
+    if not message:
+        ner_logger.exception("Message param is not passed")
+        raise KeyError("Message is required")
+
+    if not entities:
+        ner_logger.exception("Entities param is not passed")
+        raise KeyError("Entities dict is required")
+
+    if not isinstance(message, list):
+        ner_logger.exception("Message param is not in correct format")
+        raise TypeError("Message should be in format of list of string")
+
+    if not isinstance(entities, dict):
+        ner_logger.exception("Entities param is not in correct format")
+        raise TypeError("Entities should be dict of entity details")
+
+    if len(message) > 100:
+        ner_logger.exception("Maximum number of message can be 100 for "
+                             "bulk detection")
+        raise ValueError("Maximum number of message can be 100 for "
+                         "bulk detection")
+
+    if len(list(entities)) > 100:
```

Comment:
haptik_api does not have any batching on number of entities... should we add batching there ? or chatbot ner should handle ?

Reply:
I think we should have to batch at both places. Need to test for bulk entities for optimal numbers for batches.